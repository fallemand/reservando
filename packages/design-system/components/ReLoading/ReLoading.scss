@import "../../styles/variables";

.re-loading {
  position: absolute;
  right: 0;
  left: 0;
  top: 0;
  bottom: 0;
  display: flex;
  justify-content: center;
  align-items: center;

  &__image {
    width: 200px;
  }
}
