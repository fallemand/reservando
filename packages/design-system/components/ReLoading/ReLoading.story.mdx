import { Meta, Story, ArgsTable, Canvas, Description } from "@storybook/addon-docs/blocks";
import { ReLoading } from "@reservando/design-system";

<Meta title="Components/Loading" component={ReLoading} />

# ReLoading

<Canvas>
  <Story name="basic">
    {{
      components: { ReLoading },
      template: `
        <div style="position: relative">
          <ReLoading />
        </div>
      `,
    }}
  </Story>
</Canvas>

<Description of={ReLoading} />

## How to use

```html
<script lang="ts">
  import { ReLoading } from "@reservando/design-system";
</script>
```

<ArgsTable of={ReLoading} />
