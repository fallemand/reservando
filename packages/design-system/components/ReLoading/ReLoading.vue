<template>
  <div class="re-loading" :class="$attrs.class">
    <img class="re-loading__image" :src="loadingSvg" alt="loading" />
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import loadingSvg from "./loading.svg";

const ReLoading = defineComponent({
  name: "ReLoading",
  computed: {
    loadingSvg(): string {
      return loadingSvg;
    },
  },
});

export default ReLoading;
</script>

<style lang="scss" src="./ReLoading.scss" />
