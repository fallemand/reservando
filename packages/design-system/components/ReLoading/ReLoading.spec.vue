<script>
import { shallowMount } from "@vue/test-utils";
import ReLoading from "./ReLoading.vue";

describe("ReLoading component", () => {
  const slots = {
    default: "__CONTENT__",
  };

  it("should render with default props", () => {
    const wrapper = shallowMount(ReLoading, { slots });
    expect(wrapper.element).toMatchSnapshot();
  });
});
</script>
