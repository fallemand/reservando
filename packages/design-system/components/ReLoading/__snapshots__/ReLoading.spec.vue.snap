// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReLoading component should render with default props 1`] = `
<div
  class="re-loading"
>
  <img
    alt="loading"
    class="re-loading__image"
  />
</div>
`;
